### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by nimplay**
- [ ] description of #all? method
- [ ] description of #any? method
- [ ] description of #filter method

### CODE REVIEW

File: list.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require_relative 'enumerable'
+
+class MyList
+  include MyEnumerable
+
+  def init(*list)
+    @list = list
+  end
```
Comment (TedLivist) [inline list.rb]: - [ ] I get errors when I try to run your app. I think you meant `initialize`, not `init`.

  inti

File: list.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+end
```
Comment (TedLivist) [inline list.rb]: - [ ] You are supposed to implement your `#each` method in this MyList class, not use the inbuilt one in the methods in the Enumerable module. You should have something like this:

  ```
  class MyList
    include MyEnumerable
  
    def initialize(*list)
      @list = list
    end
  
    def my_each
      return to_enum(:each) unless block_given?
      counter = 0
      while counter < @list.length
        yield(@list[counter])
        counter += 1
      end
      @list
    end
  end
  ```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+module MyEnumerable
+  def list
+    @list ||= []
+  end
```
Comment (TedLivist) [inline enumerable.rb]: - [ ] We don't need this `list` item here. Remember that we define our list variable when instantiating the object, so kindly remove this one here.

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    p !arr.empty?
+  end
+
+  def filter
+    arr = []
+    list.each do |e|
+      arr.push(e) if yield e
+    end
+    p arr
+  end
```
Comment (TedLivist) [inline enumerable.rb]: - [ ] You should implement the enumerbales using the `#each` method defined in the MyList class.

- [ ] You should also return the **true** or **false**, not print a value, as this does not do much.

  Your enumerables should look something like the code snippet below. As you can see I am using the `#my_each` method I defined in the **list.rb** file
  ```
  def all?
    my_each { |item| return false unless yield item }
    true
  end

  def any?
    my_each { |item| return true if yield item }
    false
  end

  def filter
    filter_arr = []
    my_each { |item| filter_arr << item if yield item }
    filter_arr
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @nimplay and @for-ashraf,

Good job so far!
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi Team, 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**